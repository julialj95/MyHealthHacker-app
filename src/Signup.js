import React, { useState } from "react";
import { useHistory } from "react-router-dom";
import config from "./config";
import TokenService from "./services/token-service";

function Signup() {
  const [userData, setUserData] = useState({ username: "", password: "" });
  const [error, setError] = useState("");
  const history = useHistory();

  const handleChange = (e) => {
    const { name, value } = e.target;
    setUserData({
      ...userData,
      [name]: value,
    });
    console.log(userData);
  };

  const handleSignup = (e) => {
    e.preventDefault();
    const { username, password } = userData;

    fetch(`${config.API_BASE_URL}/users`, {
      method: "POST",
      headers: {
        "content-type": "application/json",
        Authorization: "Bearer " + TokenService.getAuthToken(),
      },
      body: JSON.stringify({
        username,
        password,
      }),
    })
      .then((res) => {
        console.log(res);
        if (!res.ok) {
          return res.json().then((e) => Promise.reject(e));
        }
      })
      .then(() => history.push("/login"))
      .catch((error) => setError(error));
  };

  console.log(error);

  return (
    <div className="signup">
      {" "}
      <h1>SIGN UP</h1>
      <h2 className="signup_instructions">
        Create an account to start tracking your health habits!
      </h2>
      <form className="signup_form" onSubmit={(e) => handleSignup(e)}>
        <input
          className="signup_field"
          type="text"
          name="username"
          placeholder="Create a username..."
          onChange={(e) => handleChange(e)}
        />

        <br />

        <input
          className="signup_field"
          type="text"
          name="password"
          placeholder="Create a password..."
          onChange={(e) => handleChange(e)}
        />

        <br />
        <button className="signup_btn" type="submit">
          Create Account
        </button>
      </form>
      {error ? <h2>{error.error}</h2> : null}
    </div>
  );
}

export default Signup;
