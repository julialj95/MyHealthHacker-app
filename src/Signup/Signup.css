.signup {
  text-align: center;
  margin: 10%;
  border: solid 3px #88d317;
  border-radius: 15px;
  background-color: #6e3667;
}

.signup_form {
  padding-bottom: 15px;
}

.signup_instructions {
  color: white;
  padding-left: 10px;
  padding-right: 10px;
}

.signup_header {
  color: white;
}

@media all and (min-width: 600px) {
  .signup_header {
    font-size: 26px;
  }
  .signup_instructions {
    font-size: 23px;
  }
}
@media all and (min-width: 960px) {
  .signup_header {
    font-size: 32px;
  }
  .signup_instructions {
    font-size: 28px;
  }
}
