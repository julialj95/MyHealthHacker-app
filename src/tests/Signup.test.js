import React from "react";
import ReactDOM from "react-dom";
import Signup from "../Signup";
import { MemoryRouter as Router } from "react-router-dom";
import renderer from "react-test-renderer";

it("renders without crashing", () => {
  const div = document.createElement("div");

  ReactDOM.render(
    <Router>
      <Signup />
    </Router>,
    div
  );
  ReactDOM.unmountComponentAtNode(div);
});

it("renders the UI as expected", () => {
  const tree = renderer
    .create(
      <Router>
        <Signup />
      </Router>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
