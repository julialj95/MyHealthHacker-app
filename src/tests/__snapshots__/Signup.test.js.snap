// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
<main
  className="signup"
>
   
  <h2
    className="signup_header"
  >
    SIGN UP
  </h2>
  <h3
    className="signup_instructions"
  >
    Create an account to start tracking your health habits!
  </h3>
  <form
    className="signup_form"
    onSubmit={[Function]}
  >
    <input
      className="signup_field"
      name="username"
      onChange={[Function]}
      placeholder="Create a username..."
      type="text"
    />
    <br />
    <input
      className="signup_field"
      name="password"
      onChange={[Function]}
      placeholder="Create a password..."
      type="text"
    />
    <br />
    <br />
    <button
      className="form_btn"
      type="submit"
    >
      Create Account
    </button>
  </form>
</main>
`;
